### PR TITLE
NP-32272-search-journal

### PR DESCRIPTION
**Pull request by hackon**

### CODE REVIEW

File: docs/open-api.yaml [hunk 1]
```diff
@@ -248,6 +314,27 @@ components:
             $ref: '#/components/schemas/Problem'
 
   schemas:
+    PaginatedJournal:
```
Comment (monaullah) [inline docs/open-api.yaml:317]: PaginatedJournalHits?

File: src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+        var pagesSearchResult = objectMapper.readValue(response.getBody(),
+                new TypeReference<PaginatedSearchResult<JournalResult>>() {
+                });
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+        assertThat(pagesSearchResult.getHits(), containsInAnyOrder(expectedSearchResult.getHits().toArray()));
+    }
+
+    @Test
+    void shouldReturnResultWithSuccessWhenQueryIsName() throws IOException, UnprocessableContentException {
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java:148]: Big test - extract some of it? Setup method?

File: src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+        assertThat(pagesSearchResult.getTotalHits(), is(equalTo(dataportenSearchResult.size())));
+        var expectedSearchresult = getExpectedPaginatedSearchResultNameSearch(
+                dataportenSearchResult, year, name, 0, 10);
+        assertThat(pagesSearchResult.getHits(), containsInAnyOrder(expectedSearchresult.getHits().toArray()));
+    }
+
+    @Test
+    void shouldReturnResultWithSuccessWhenQueryIsNameAndOffsetIs10() throws IOException, UnprocessableContentException {
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java:178]: Big test - extract some of it? Setup method?

File: src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+
+        var problem = response.getBodyObject(Problem.class);
+
+        assertThat(problem.getDetail(), is(containsString("Offset")));
+
+    }
+
+    @Test
+    void shouldLogAndReturnBadGatewayWhenChannelClientReturnsUnhandledResponseCode() throws IOException {
+
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java:296]: Big test - extract some of it? Setup method?

File: src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java (hunk 2 above)
Comment (hackon) [reply src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java:147]: The setup is kind of clunky. I did some cleaning, but most of it "has" to stay.

**Review by monaullah — APPROVED** [review]

**Pull request merged**